homeassistant/components/mqtt/client.py

```diff
@@ -688,6 +691,29 @@ def _process_client_subscriptions() -> list[tuple[int, int]]:
         if errors:
             _raise_on_errors(errors)
 
+    async def _async_perform_unsubscribes(self) -> None:
+        """Perform pending MQTT client unsubscribes."""
+        if not self._pending_unsubscribes:
+            return
+
+        topics = set(self._pending_unsubscribes)
+        self._pending_unsubscribes = set()
+
+        def _process_client_unsubscribes() -> tuple[int, int]:
+            """Initiate all subscriptions on the MQTT client and return the results."""
+            result, mid = self._mqttc.unsubscribe(list(topics))
+            return result, mid
+
+        async with self._paho_lock:
+            result, mid = await self.hass.async_add_executor_job(
+                _process_client_unsubscribes
```

Comment:
Just like in https://github.com/home-assistant/core/pull/92172#pullrequestreview-1413697185, you can now get rid of `_process_client_unsubscribes()`

ie.:
```
    result, mid = await self.hass.async_add_executor_job(
        self._mqttc.unsubscribe, list(topics)
    )
```

Reply:
I know, @emontnemery still has a look at the other PR first,  I'll follow up after merge and rebate. We are in a busy period where some other fixes have priority.